src/app/tests/AppTestContext.cpp

```diff
@@ -49,6 +49,12 @@ void AppContext::SetUpTestSuite()
 
 void AppContext::TearDownTestSuite()
 {
+    // Adding a DrainAndServiceIO call fixes the teardown for some tests (TestAclAttribute and TestReportScheduler); these were
+    // triggering failures in tests that follow them when running on nRF CI (Zephyr native_posix where all Unit Tests are compiled
+    // into a single file)
+    // TODO: understand this more and solve underlying issue
+    LoopbackMessagingContext::DrainAndServiceIO();
```

Comment:
Is this required as part of this change?  Seems like an unrelated fix that could use very careful review not in the context of a mass-change....

Reply:
well, I only need it in order to be able to run `TestAclAttribute.cpp` and `TestTestReportScheduler.cpp` before the other tests. 
I have just put it in a seperate PR: https://github.com/project-chip/connectedhomeip/pull/33654